### PR TITLE
Fix MD5 FIPS compliance

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**

### CODE REVIEW

File: src/Integration/Suppression/ManagedMD5.cs [hunk 1]
```diff
@@ -0,0 +1,461 @@
+﻿/* Copyright (C) David Anson
+ * https://dlaa.me/blog/post/9380245
```
Comment (duncanp-sonar) [inline src/Integration/Suppression/ManagedMD5.cs:2]: What, if anything, needs to go into the `THIRD-PARTY-NOTICES` file?

File: src/Integration/Suppression/ChecksumCalculator.cs [hunk 2]
```diff
@@ -41,7 +40,7 @@ public static string Calculate(string text)
             }
 
             string content = Regex.Replace(text, "\\s", ""); // strip whitespace
-            using (MD5 md5Digest = MD5.Create())
+            using (var md5Digest = new Md5Managed())
```
Comment (duncanp-sonar) [inline src/Integration/Suppression/ChecksumCalculator.cs:43]: As previously discussed, what validation do you think we need to do to satisfy ourselves that the algorithm is is correctly implemented?

Comment (rita-g-sonarsource) [reply src/Integration/Suppression/ChecksumCalculator.cs:43]: We are using it for suppression matching so I think we should focus on that. I tried C++/C# suppressions, long lines, short lines, variables, comments. I'll try unicode characters. Anything else do you think we should test?

File: src/Integration/Suppression/ManagedMD5.cs (hunk 1 above)
Comment (rita-g-sonarsource) [reply src/Integration/Suppression/ManagedMD5.cs:2]: I don't see a copyright in his project/site that I can point to. 
How about this:
```
License notice for MD5 Algorithm
-----------------------------------------------------
Copyright (C) David Anson 
https://dlaa.me/blog/post/9380245
```

File: src/Integration/Suppression/ChecksumCalculator.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/Integration/Suppression/ChecksumCalculator.cs:43]: Those sound like the right sort of things to test. I was expecting to see some unit tests comparing the results against the .NET implementation.
Did you consider using an existing NuGet package like https://www.nuget.org/packages/BouncyCastle/ or https://www.nuget.org/packages/CryptSharpOfficial/ that have seem to be popular?

File: src/Integration/Suppression/ManagedMD5.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/Integration/Suppression/ManagedMD5.cs:2]: Seems ok. What about the RSA license notice from the comments in lines 10-44?

**Pull request closed without merging**